src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java

```diff
@@ -938,7 +938,7 @@ public boolean hasPermission(final UUID id, @NotNull final Action action)
         public boolean hasPermission(final Rank rank, @NotNull final Action action)
         {
             return (rank == Rank.OWNER && action != Action.GUARDS_ATTACK)
-                     || Utils.testFlag(permissions.get(rank), action.getFlag());
+                     || (permissions != null && action != null && rank != null && Utils.testFlag(permissions.get(rank), action.getFlag()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2589)